packages/datadog-plugin-aws-sdk/src/aws_helpers.js

```diff
@@ -0,0 +1,191 @@
+'use strict'
+
+const awsHelpers = {
+  wrapCallback (tracer, span, callback, parent, config) {
+    return function (err, response) {
+      // "this" should refer to response object https://github.com/aws/aws-sdk-js/issues/781#issuecomment-156250427
+      awsHelpers.addAwsResponseTags(span, this)
+
+      config.hooks.addCustomTags(span, this.request.params)
+      awsHelpers.finish(span, err)
+
+      if (typeof callback === 'function') {
+        return tracer.scope().activate(parent, () => callback.apply(null, arguments))
+      }
+    }
+  },
+
+  finish (span, err) {
+    if (err) {
+      span.addTags({
+        'error.type': err.name,
+        'error.msg': err.message,
+        'error.stack': err.stack
+      })
+
+      if (err.requestId) {
+        span.addTags({ 'aws.request_id': err.requestId })
+      }
+    }
+
+    span.finish()
+  },
+
+  addAwsRequestTags (span, request, operation, params, serviceName) {
+    if (span) {
+      if (request && request.httpRequest && context.request.httpRequest.endpoint) {
+        span.addTags({ 'aws.url': request.httpRequest.endpoint.href })
+      }
+
+      // sync with serverless on how to normalize to fit existing conventions
+      // <operation>_<specialityvalue>
+      this.addAwsServiceTags(span,
+        operation,
+        params,
+        serviceName
+      )
+    }
+  },
+
+  addAwsResponseTags (span, response) {
+    if (span) {
+      if (response.requestId) {
+        span.addTags({ 'aws.request_id': response.requestId })
+      }
+
+      // status code and content length to match what serverless captures
+      if (response.httpResponse) {
+        if (response.httpResponse.headers && response.httpResponse.headers['content-length']) {
+          span.addTags({ 'http.content_length': response.httpResponse.headers['content-length'].toString() })
+        }
+
+        if (response.httpResponse.statusCode) {
+          span.addTags({ 'http.status_code': response.httpResponse.statusCode.toString() })
+        }
+      }
+
+      // SNS.createTopic is invoked with name but returns full arn in response data
+      // which is used elsewhere to refer to topic
+      if (response.data && response.data.TopicArn && response.request.operation) {
+        span.addTags({ 'aws.sns.topic_arn': response.data.TopicArn })
+        span.addTags({ 'resource.name': `${response.request.operation} ${response.data.TopicArn}` })
+      }
+
+      // TODO: should arn be sanitized or quantized in some way here,
+      // for example if it contains a phone number?
+    }
+  },
+
+  // names are inconsistently prefixed with AWS or Amazon
```

Comment:
Fwiw this was just something the serverless folks mentioned in RFC comments, i haven't really vetted this fully. Seems like a very aws thing to do though. 

Assuming the inconsistency is frequent...In my mind, Given that this instrumentation is going to split out services by aws service, it would seem like we want to offer a way for users to also easily _group by_ related aws services. Searching in `traces/App analytics` allows things like [wildcarding](https://docs.datadoghq.com/logs/explorer/search/#wildcards) so adding some consistency to the prefix would make this much easier.

I also realize, at a high level, that tracing instrumentation shouldn't necessarily be written to fit an arbitrary UI convention, and there's also  probably _other_ indirect ways of grouping by aws service in the UI (say, searching for only traces with an `aws.agent` tag), but this just seemed like a low hanging fruit way to make things a little nicer. Happy to axe if it's more hassle than it's worth.

Reply:
As discussed yesterday, there is no prefix anymore since we use a different field.